### ISSUE TITLE
setx should allow reference to colnames in data

### ISSUE THREAD

**Issue by izahn** [IQSS/Zelig#199]
Currently `setx` does not handle transformation well. I would expect this to work:

```r
library(Zelig)
m <- zls$new()
m$zelig(speed ~ poly(dist, 2), data = cars)
m$setx(dist = 15)

## Error in poly(dist, 2) : 
##   anyNA() applied to non-(list or vector) of type 'closure'
## In addition: Warning message:
## In reduce(dataset = "MEANINGLESS ARGUMENT", s, formula = f2, data = .,  :
##   Variable 'dist' not in data set.
```

**Comment by christophergandrud** [comment]
This is partially related to https://github.com/IQSS/Zelig/issues/156 where `sim` fails with factor variables. 

The `reduce` utility seems to need a lot of work.

**Comment by christophergandrud** [comment]
Factors are resolved in v5.0-15. 

Polynomials and splines (https://github.com/IQSS/Zelig/issues/218) are next.

**Comment by christophergandrud** [comment]
Dev version supports the following syntax which gets the results you want, e.g.:

```r
library(Zelig)
library(dplyr)

# Create squared dist
cars.poly <- cbind(cars, I(cars$dist^2))
names(cars.poly)[3] <- 'dist_2'

# Define range of fitted dist values
dist_range <- min(cars.poly$dist):max(cars.poly$dist)

# Estimate, simulate, and plot
cars_set <- zelig(speed ~ dist + dist_2, data = cars.poly, model = 'ls', cite = FALSE) %>%
    setx(dist = dist_range, dist_2 = dist_range^2) %>%
    sim() %>%
    plot
```

![image](https://cloud.githubusercontent.com/assets/1285805/23704892/52838aa6-0407-11e7-9b30-438e66aed0fd.png)


Not the streamlined syntax we want yet. But the model type is at least possible. 

Closing the issue for the time being, but will be back to this for a streamlined syntax.
